File: models/team.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class Team:
    club_id: str
    name: str
    short_name: str
    league_code: str
    division_id: str
    season: str
    gender: Optional[str] = None
    format: Optional[str] = None
    id: Optional[int] = None
    followers_count: Optional[int] = 0
    active: bool = True
    created_at: Optional[datetime] = None
    last_update: Optional[datetime] = None

File: src/types/Team.ts
import { Club } from "./Club";
import { Division } from "./Division";
import { EnumFormat } from "./enums/Format";
import { EnumGender } from "./enums/Gender";
import { EnrichedPoolDTO, Pool } from "./Pool";

export interface Team {
    id: number;
    clubId: string;
    name: string;
    shortName: string;
    leagueCode: string;
    divisionId: number;
    format: EnumFormat;
    gender: EnumGender;
    season: string;
    followersCount: number;
    active: boolean;
    createdAt: string;
    lastUpdate: string;
}

export interface TeamWithStats {
    id: number;
    name: string;
    shortName: string;
    format: EnumFormat;
    gender: EnumGender;
    season: string;
    followersCount: number;
    logoUrl: string | null;
    points: number;
    played: number;
    wins: number;
    losses: number;
    pointsPenalty: number;
    coefSets: number;
    coefPoints: number;
}

export interface EnrichedTeamDTO {
    id: number;
    name: string;
    shortName: string;
    format: EnumFormat;
    gender: EnumGender;
    season: string;
    followersCount: number;
    division: Division;
    club: Club
    pools: EnrichedPoolDTO[];
}